dd-sdk-android-core/src/main/kotlin/com/datadog/android/v2/core/SdkInternalLogger.kt

```diff
@@ -30,38 +30,44 @@ internal class SdkInternalLogger(
 ) : InternalLogger {
 
     /**
-     * Global Dev Logger. This logger is meant for user's debugging purposes.
+     * This logger is meant for user's debugging purposes.
      * Logcat logs are conditioned by the [Datadog.libraryVerbosity].
      * No Datadog logs should be sent.
      */
-    internal val devLogger = devLogHandlerFactory.invoke()
+    internal val userLogger = userLogHandlerFactory.invoke()
 
     /**
-     * Global SDK Logger. This logger is meant for internal debugging purposes.
+     * This logger is meant for internal debugging purposes.
      * Logcat logs are conditioned by a BuildConfig flag (set to false for releases).
      */
-    internal val sdkLogger = sdkLogHandlerFactory.invoke()
+    internal val maintainerLogger = maintainerLogHandlerFactory.invoke()
+
+    private val onlyOnceUserMessages = mutableSetOf<String>()
+    private val onlyOnceMaintainerMessages = mutableSetOf<String>()
+    private val onlyOnceTelemetryMessages = mutableSetOf<String>()
```

Comment:
do you want to keep the strings or maybe a hashcode will be lighter with the caveat that you may have some collisions ? I wonder in case of the String if we should add a backup logic here in case of high cardinality in logs (I am thinking of sending a log with an id every time an operation is performed)

Reply:
We don't have that many logs that need to be sent only once, so I think using String won't have a bad impact.